apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/features/TestFeatures.java

```diff
@@ -42,13 +45,31 @@ public static void initialize(Activity parentActivity) {
         sTestFeatureModels.add(new TestFeatureTitle(R.string.crashes_title));
         sTestFeatureModels.add(new TestFeature(R.string.title_crashes, R.string.description_crashes, CrashActivity.class));
         sTestFeatureModels.add(new TestFeature(R.string.title_error, R.string.description_error, ManagedErrorActivity.class));
+        sTestFeatureModels.add(new TestFeature(R.string.title_had_memory_warning, hadMemoryWarning() ? R.string.description_had_memory_warning : R.string.description_didnt_have_memory_warning, new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // do not handle the click.
+            }
+        }));
         sTestFeatureModels.add(new TestFeatureTitle(R.string.miscellaneous_title));
         sTestFeatureModels.add(new TestFeature(R.string.title_custom_properties, R.string.description_custom_properties, CustomPropertiesActivity.class));
         sTestFeatureModels.add(new TestFeature(R.string.title_device_info, R.string.description_device_info, DeviceInfoActivity.class));
         sTestFeatureModels.add(new TestFeatureTitle(R.string.title_data));
         sTestFeatureModels.add(new TestFeature(R.string.title_data, R.string.description_data, DataActivity.class));
     }
 
+    @SuppressWarnings("unchecked")
+    private static boolean hadMemoryWarning() {
+        // TODO: Replace with return Crashes.hadMemoryWarningInLastSession();
+        try {
+            @SuppressWarnings("unchecked") final Class<? extends AppCenterService> crashes = (Class<? extends AppCenterService>) Class.forName("com.microsoft.appcenter.crashes.Crashes");
+            final Method hadMemoryWarning = crashes.getMethod("hadMemoryWarningInLastSession");
+            return ((AppCenterFuture<Boolean>) hadMemoryWarning.invoke(null)).get();
```

Comment:
Can we avoid block UI here?
BTW Why do we need async getter and `AppCenterFuture`?

Reply:
https://github.com/microsoft/appcenter-sdk-android/pull/1216/files#diff-f948cee626f7c221be3d4eee50bedcb8R340